utils.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sun Apr 16 19:12:51 2023
+
+@author: Sumedh Prasad
+
+utils code
+"""
+
+from typing import List, Tuple
+import os
+from os.path import join, dirname
+import logging
+
+from pathlib import Path
+import mysql.connector as mysql
+from dotenv import load_dotenv,dotenv_values
+
+os.chdir(Path("E:/E2E_DTP_Project"))
+
+
+logging.basicConfig(filename='myapp.log',
+                    format='%(asctime)s %(message)s', 
+                    level=logging.INFO)
+
+env = Path(".env")
+load_dotenv(dotenv_path=env)
+
+from dotenv import load_dotenv
+load_dotenv()
+
+def connect_db(host: str, user: str) -> Tuple[mysql.connection.MySQLConnection, str]:
+    """this function connects to mysql database and returns conn, cur as tuple
+
+    Args:
+        host (str): pass host name
+        user (str): pass user name
+
+    Returns:
+        Tuple[mysql.connection.MySQLConnection, str]: 
+    """
+    try:
+        logging.info('Started')
+        conn = mysql.connect(host=host,
+                            user=user,
+                            password="12345")
+        cur = conn.cursor()
+        logging.info("Connected to MySQL successfully!")
+        logging.info('ended')
+    except Exception as e:
+        logging.debug(e)
+    return conn, cur
+
+conn, cur = connect_db(host="localhost", user="root")
+
+import pandas as pd
+import numpy as np
```

Comment:
all imports should go at the start of the script

Reply:
Changes done